File: codegen/results/v001/qwen3.5_4b_nothink/20260408_194201/go/003_fibonacci/solution.go
package main

import (
	"fmt"
	"os"
)

func main() {
	// Get N from command line arguments
	n, err := parseInt(os.Args[1])
	if err != nil {
		fmt.Fprintln(os.Stderr, "Error:", err)
		os.Exit(1)
	}

	// If N < 1, print nothing
	if n < 1 {
		return
	}

	// Generate Fibonacci numbers up to N
	a, b := 1, 1
	count := 0
	for {
		if a > n {
			break
		}
		fmt.Println(a)
		count++
		if b > n {
			break
		}
		fmt.Println(b)
		count++
		// Check if b exceeds n to prevent overflow in next iteration if N is large
		if a+n > n {
			break
		}
		a, b = b, a+b
	}
}

func parseInt(s string) (int, error) {
	var n int
	_, err := fmt.Sscanf(s, "%d", &n)
	if err != nil {
		return 0, err
	}
	return n, nil
}

File: codegen/results/v001/qwen3.5_4b_nothink/20260408_194201/typescript/003_fibonacci/solution.ts
import { argv } from 'process';

// Read N from command line argument
const N = parseInt(argv[2], 10);

// Check if N is a valid positive integer
if (Number.isNaN(N) || N < 1) {
  // If N < 1 or invalid, print nothing
  process.exit(0);
}

// Generate Fibonacci numbers up to N
let a = 1;
let b = 1;

// Print the first two numbers (1, 1) if they are within range
if (a <= N) {
  console.log(a);
}
if (b <= N) {
  console.log(b);
}

// Generate subsequent Fibonacci numbers
while (true) {
  const next = a + b;
  
  // If next is greater than N, stop
  if (next > N) {
    break;
  }
  
  console.log(next);
  
  // Update a and b for the next iteration
  a = b;
  b = next;
}